cloud/shared/bin/lib/config_loader.py

```diff
@@ -63,27 +64,34 @@ def get_env_variables(self, variable_definitions: dict):
     # TODO: we do not validate type of the variable as we only have
     # strings currently. If we add non-strings, will need to validate
     def _validate_config(self, variable_definitions: dict, configs: dict):
-        is_valid = True
         validation_errors = []
 
         for name, definition in variable_definitions.items():
-            is_required = definition.get("required", False)
+            is_required = definition.get('required', False)
             config_value = configs.get(name, None)
 
             if is_required and config_value is None:
-                is_valid = False
-                validation_errors.append(
-                    f"{name} is required, but not provided")
+                validation_errors.append(f'[{name}] required, but not provided')
 
-            is_enum = definition.get("type") == "enum"
+            is_enum = definition.get('type') == 'enum'
 
             if config_value is not None and is_enum:
-                if config_value not in definition.get("values"):
-                    is_valid = False
+                if config_value not in definition.get('values'):
                     validation_errors.append(
-                        f"{config_value} not supported enum for {name}")
+                        f'[{name}] \'{config_value}\' is not a supported enum value. Want a value in [{", ".join(definition.get("values"))}]'
+                    )
 
-        return is_valid, validation_errors
+            value_regex = definition.get('value_regex', None)
+            if config_value is not None and value_regex:
+                validation_error = definition.get(
+                    'value_regex_error_message', None)
+                if not validation_error:
```

Comment:
why is this **not** validation_error? I'd imagine if validation error is empty we're good?

Reply:
Renamed to `validation_error_message`. This is saying that we'll throw an error if there's no `validation_error_message` value defined.

This is actually something that shouldn't happen assuming that the validate_variable_definitions test has run on the config. As part of #2887, we should be able to remove this entirely. I left a TODO to that effect.